lib/Sema/TypeCheckEffects.cpp

```diff
@@ -1424,6 +1424,8 @@ class Context {
     DiagnoseErrorOnTry = b;
   }
 
+  SourceLoc awaitLoc = SourceLoc();
```

Comment:
Might be good to comment what this member represents: the location of the innermost `await` during the walk, for diagnostic purposes.

Reply:
Done!